### PR TITLE
[incubator-kie-drools-5724] [new-parser] Broken ExisistentialTest#tes…

### PR DESCRIPTION
**Pull request by tkobayas**
…t2NotsWithAnd

**Issue**:

* https://github.com/apache/incubator-kie-drools/issues/5724

### Before PR in `drools-model-codegen`
```
[ERROR] Tests run: 2434, Failures: 76, Errors: 0, Skipped: 9
```

### After PR in `drools-model-codegen`
```
[ERROR] Tests run: 2434, Failures: 74, Errors: 0, Skipped: 9
```

Also fixed some errors in `kie-dmn-validation` build.

### CODE REVIEW

**Comment by tkobayas** [comment]
@yurloc @mariofusco @gitgabrio Please review, thanks!

**Review by mariofusco — APPROVED** [review]

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 [hunk 1]
```diff
@@ -123,6 +123,7 @@ lhsUnary : (
            | lhsEval consequenceInvocation*
            | lhsForall
            | lhsAccumulate
+           | LPAREN lhsExpression RPAREN namedConsequenceInvocation?
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:138]: @tkobayas I confess that I do not 100% understand how exactly the changes in this PR work. In an attempt to better understand how individual changes contribute to the fix, I experimented and reverted parts of the PR.

I _accidentally_ found out this specific addition appears to be redundant. Everything seems to work as expected even without this line. Please reconsider this addition and see if you can find a good reason to remove it. If uncertain, feel free to keep it there.

**Review by yurloc — APPROVED** [review]
I don't understand this one fully but it works fine and fixes the problem. One small comment, otherwise LGTM!

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 (hunk 1 above)
Comment (mariofusco) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:138]: Are we testing named consequences already?

Comment (tkobayas) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:138]: @yurloc Thank you for pointing out that. I had thought that `MiscDRLParserTest.namedConsequenceAfterEnclosed` fails without `LPAREN lhsExpression RPAREN namedConsequenceInvocation?`, but indeed it passes without the line. Now I added a new test `namedConsequenceOrWithBindVariables` which requires the line.

Comment (tkobayas) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:138]: > Are we testing named consequences already?

Yes, `MiscDRLParserTest` contains a bunch of named consequence tests.

Comment (yurloc) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4:138]: > Now I added a new test `namedConsequenceOrWithBindVariables` which requires the line.

Excellent, thanks!

**Comment by tkobayas** [comment]
```
2024-04-30T06:17:36.8286605Z [INFO] --- surefire:3.1.2:test (default-test) @ drools-model-codegen ---
...
2024-04-30T06:18:54.0508349Z [WARNING] Tests run: 2434, Failures: 0, Errors: 0, Skipped: 9
```

**Pull request merged**